include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,405 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+#pragma once
+
+#include <cassert>
+#include "../__p0009_bits/dynamic_extent.hpp"
+#include "../__p0009_bits/extents.hpp"
+#include "../__p0009_bits/mdspan.hpp"
+#include "../__p0009_bits/layout_left.hpp"
+#include "../__p0009_bits/layout_right.hpp"
+#include "../__p0009_bits/layout_stride.hpp"
+
+namespace std {
+namespace experimental {
+
+template <size_t padding_stride>
+struct layout_left_padded;
+
+template <size_t padding_stride>
+struct layout_right_padded;
```

Comment:
These guys need to go before any of the other layout mappings are actually done. Also the order of include matters for the above stuff. (layout_stride actually needs to go first). The (not really) forward declaration needs to look like this:
```c++
template<size_t padding_stride>
struct layout_left_padded {
  template<class Extents>
    class mapping;
};
```

I think.

Reply:
That should have been the other question I asked about adding all those converting constructors lol ; - P

Reply:
Fixed. I put it in a separate header